src/main/resources/avro/bdg.avdl

```diff
@@ -57,8 +57,55 @@ record Contig {
   union { null, string } species = null;
 }
 
+record Sequence {
+  union { null, string } bases;
+  union { null, string } qualities;
+}
+
+/**
+   The DNA fragment that is was targeted by the sequencer, resulting in
+   one or more reads.
+*/
+record Fragment {
+  /**
+     The name of this Fragment.
+  */
+  union { null, string } readName = null;
```

Comment:
How do you envision `readName` to work?

Separately, maybe we need a `barcodeId` or some other type of identifier, for more general cases. I am thinking of cases where you perform library prep that does some kind of subassembly (e.g., [1]).  Or droplet-seq.

[1]: http://www.nature.com/nmeth/journal/v7/n2/full/nmeth.1416.html